contracts/Strategy.sol

```diff
@@ -53,80 +222,624 @@ contract Strategy is BaseStrategy {
+                wantToDeposit,
+                0
+            );
+    }
+
+    function transferBento(address to, uint256 shares) internal {
+        bentoBox.transfer(
+            BIERC20(address(want)),
+            address(this),
+            address(to),
+            shares
+        );
+    }
+
+    function liquidateKashiPair(uint256 kashiPairIndex, uint256 sharesToFree)
+        internal
+        returns (uint256 _shareLiquidated)
+    {
+        // We need to call accrue to accurately calculate totalAssets
+        accrueInterest(kashiPairIndex);
+
+        uint256 liquidShares = kashiPairLiquidShares(kashiPairIndex);
+        if (sharesToFree > liquidShares) {
+            sharesToFree = liquidShares;
+        }
+
+        if (sharesToFree == 0) return 0;
+
+        uint256 fractionsToFree =
+            bentoSharesToKashiFraction(
+                kashiPairs[kashiPairIndex].kashiPair,
+                sharesToFree,
+                true
+            );
+
+        // Remove from masterChef if there is a non-zero pid
+        if (kashiPairs[kashiPairIndex].pid != 0) {
+            uint256 fractionInMc = kashiFactionInMasterChef(kashiPairIndex);
+            uint256 fractionsToFreeFromMc = fractionsToFree;
+            if (fractionsToFreeFromMc > fractionInMc) {
+                fractionsToFreeFromMc = fractionInMc;
+            }
+            masterChef.withdraw(
+                kashiPairs[kashiPairIndex].pid,
+                fractionsToFreeFromMc
+            );
+        }
+
+        uint256 fractionBalance = kashiFractionInPair(kashiPairIndex);
+
+        if (fractionsToFree > fractionBalance) {
+            fractionsToFree = fractionBalance;
+        }
+
+        _shareLiquidated = kashiPairs[kashiPairIndex].kashiPair.removeAsset(
+            address(this),
+            fractionsToFree
+        );
+
+        // Redeposit into the masterChef if there's some spare change
+        depositKashiInMasterChef(kashiPairIndex);
+    }
+
+    // sell all function
+    function sell() internal {
+        uint256 sushiBal = balanceOfSushi();
+        if (sushiBal == 0) {
+            return;
+        }
+
+        sushiRouter.swapExactTokensForTokens(
+            sushiBal,
+            uint256(0),
+            path,
+            address(this),
+            now
+        );
+    }
+
+    function accrueInterest(uint256 kashiPairIndex) internal {
+        (, uint256 lastAccrued, ) =
+            kashiPairs[kashiPairIndex].kashiPair.accrueInfo();
+        // Accure interest
+        if (block.timestamp > lastAccrued) {
+            kashiPairs[kashiPairIndex].kashiPair.accrue();
+        }
+    }
+
+    function setDustThreshold(uint256 _newDustThreshold)
+        external
+        onlyAuthorized
+    {
+        dustThreshold = _newDustThreshold;
+    }
+
+    function setPath(address[] calldata _path) external onlyGovernance {
+        path = _path;
+    }
+
+    function setRouter(address _router) external onlyGovernance {
```

Comment:
left a comment in issue about the rules on router and path being hardcoded if possible, unless theres a good reason why this cant be, since management may need to switch path easily, and governance is not as fast.

And we don't allow address injection outside governance powers